src/pipeline/server.rs

```diff
@@ -11,16 +13,20 @@ use futures::{Future, Poll, Async};
 
 /// A server `Task` that dispatches `Transport` messages to a `Service` using
 /// protocol pipelining.
-pub struct Server<S, T>
-    where S: ServerService,
-          T: Transport,
+pub struct Server<S, T, B>
+    where T: Transport,
+          S: Service<Request = Message<T::Out, Body<T::BodyOut, T::Error>>,
```

Comment:
Ok, let's move discussion to https://github.com/tokio-rs/tokio-proto/issues/64, not too relevant for this particular PR anyway!